### ISSUE TITLE
fabric8 docker-maven-plugin does not handle proper shutdown of Postgres

### ISSUE THREAD

**Issue by Karm** [quarkusio/quarkus-quickstarts#118]
Hello,

Apparently if one runs more than one test that uses Postgres in succession, it does not land well, see [Logs](#Logs).

We have been successfully using Github's [Docker integration for allocating and deallocating Postgres not only as a Docker container](https://github.com/web-servers/db-allocator/blob/master/src/main/java/org/jboss/qa/db/PostgreContainerAllocator.java#L56).
It sports 
 * active waiting for TCP
 * and using [SELECT 1](https://github.com/web-servers/db-allocator/blob/master/src/main/java/org/jboss/qa/db/Allocator.java#L138) to test that the DB system is really up
 * it tears down and terminates the container correctly

The dynamic loading of DB driver would be likely a huge overkill, especially for native image :smiley: perhaps, but the idea of driving the allocation from the integration test code is IMHO worth pondering. It might prove more stable than relying on fabric8 maven plugin and 
```
<wait>
    <tcp>
        <mode>mapped</mode>
        <ports>
            <port>5432</port>
        </ports>
    </tcp>
    <time>10000</time>
</wait>
```

# Logs

```
[ERROR] DOCKER> Error occurred during container startup, shutting down...
[ERROR] DOCKER> I/O Error [Unable to start container id [bb1a9c658c61] : {"message":"driver failed programming external connectivity on endpoint postgres-2 (91f5f0141ebfad562ef9aaca29e8a47fd05872a6fb9822a437ac13ca43e26363): Bind for 0.0.0.0:5432 failed: port is already allocated"} (Internal Server Error: 500)]
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] application-configuration .......................... SUCCESS [ 54.645 s]
[INFO] application-lifecycle-events ....................... SUCCESS [ 56.919 s]
[INFO] quarkus-quickstart ................................. SUCCESS [ 56.146 s]
[INFO] quarkus-quickstart-async ........................... SUCCESS [ 56.648 s]
[INFO] hibernate-orm-resteasy ............................. SUCCESS [02:35 min]
[INFO] hibernate-orm-panache-resteasy ..................... FAILURE [ 17.110 s]
[INFO] infinispan-client .................................. SKIPPED
[INFO] validation ......................................... SKIPPED
[INFO] scheduling-periodic-tasks .......................... SKIPPED
[INFO] using-websockets ................................... SKIPPED
[INFO] using-opentracing .................................. SKIPPED
[INFO] using-spring-di .................................... SKIPPED
[INFO] using-jwt-rbac ..................................... SKIPPED
[INFO] rest-client ........................................ SKIPPED
[INFO] kafka-quickstart ................................... SKIPPED
[INFO] rest-json .......................................... SKIPPED
[INFO] quarkus-quickstart-knative ......................... SKIPPED
[INFO] quarkus-quickstart-testing ......................... SKIPPED
[INFO] microprofile-fault-tolerance ....................... SKIPPED
[INFO] microprofile-health ................................ SKIPPED
[INFO] microprofile-metrics ............................... SKIPPED
[INFO] quickstart-projects ................................ SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 06:37 min

[ERROR] Failed to execute goal io.fabric8:docker-maven-plugin:0.28.0:start (docker-start) on project hibernate-orm-panache-resteasy: I/O Error: Unable to start container id [bb1a9c658c61] : {"message":"driver failed programming external connectivity on endpoint postgres-2 (91f5f0141ebfad562ef9aaca29e8a47fd05872a6fb9822a437ac13ca43e26363): Bind for 0.0.0.0:5432 failed: port is already allocated"} (Internal Server Error: 500) -> [Help 1]
```
```
karm@local:~/workspaceRH/quarkus-quickstarts (master=)$ docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS                    NAMES
96d8e3f6dc28        postgres:10.5       "docker-entrypoint.s…"   19 minutes ago      Up 19 minutes       0.0.0.0:5432->5432/tcp   postgres-1
```

**Comment by Karm** [comment]
Probably resolved in #153  #154

**Comment by cescoffier** [comment]
will be fixed with #213

**Comment by gastaldi** [comment]
This issue can be closed, I suppose?
